### DISCUSSION TITLE
Supporting modular programming (via plugin system)

### DISCUSSION THREAD

**Discussion by darron1217** [strapi/strapi#7808] [Legacy Topics]
Firstly, I'm a big fan of strapi. (Strapi Team is doing great job👍 )

As the system getting bigger, I had to organize models with group.

So I tried to solve it by using **local plugin**. But I realized that the two commands below works diffrently.

`strapi generate:api Test` // This creates REST API and GraphQL queries automatically
`strapi generate:api Test --plugin my-plugin` // This creates only models and empty controller, service (without routes)

I wonder if is intended behavior.
If then, I wonder why

P.S. If there's a way to create group in `./api` folder, that would be alternative solution for me :)

**Comment by derrickmehaffy** [comment]
I can certainly agree, I'm not sure if it's within our roadmap yet but I will ping @Aurelsicoko and @JabStrapi to let them know :)
